src/main/java/org/apache/maven/plugins/jar/AbstractJarMojo.java

```diff
@@ -262,7 +294,7 @@ public File createArchive() throws MojoExecutionException {
                     getLog().warn("JAR will be empty - no content was marked for inclusion!");
                 }
             } else {
-                archiver.getArchiver().addDirectory(contentDirectory, getIncludes(), getExcludes());
+                archiver.getArchiver().addFileSet(getFileSet(contentDirectory));
```

Comment:
Maybe we can simple add next method `addDirectory` to Archiver with `addDefaultExcludes`?